operator/controllers/lifecycle/keptntaskdefinition/controller.go

```diff
@@ -58,13 +58,38 @@ func (r *KeptnTaskDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.
 		r.Log.Error(err, "Failed to get the KeptnTaskDefinition")
 		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 	}
+	defSpec := common.GetRuntimeSpec(definition)
+	if definition.Spec.Container == nil && defSpec != nil { //if the spec is well-defined
 
-	if !reflect.DeepEqual(definition.Spec.Function, klcv1alpha3.FunctionSpec{}) {
-		err := r.reconcileFunction(ctx, req, definition)
+		// get configmap reference either existing configmap name or inline generated one
+		cmName := r.getCmName("keptnfn-"+definition.Name, defSpec)
```

Comment:
using `keptnfn-<definition.Name>` could lead to problems if the TaskDefinition Name is >= 254 chars long - can we use the `TruncateString` helper function to limit the name length of the function CM?